src/script/common/c_content.cpp

```diff
@@ -559,7 +559,20 @@ void read_content_features(lua_State *L, ContentFeatures &f, int index)
 
 	f.drawtype = (NodeDrawType)getenumfield(L, index, "drawtype",
 			ScriptApiNode::es_DrawType,NDT_NORMAL);
-	getfloatfield(L, index, "visual_scale", f.visual_scale);
+
+	lua_getfield(L, index, "visual_scale");
+
+	if(lua_isnumber(L, -1))
+	{
+		f32 visual_scale_f;
+		getfloatfield(L, index, "visual_scale", visual_scale_f);
+
+		f.visual_scale *= visual_scale_f;
```

Comment:
This seems like it won't work: `f.visual_scale` should be zero-initialized here, so `f.visual_scale * visual_scale_f` should still be a zero vector. I recommend making this an assignment `f.visual_scale = v3f(visual_scale_f)`.

Reply:
I don't think so. Default `f.visual_scale` is initialized as `v3f(1.0f)` in `ContentFeatures::reset()` method.

Reply:
And has `reset` necessarily been called? What I propose is more readable.

Reply:
Done